oml/utils/download_mock_dataset.py

```diff
@@ -19,54 +23,98 @@
 from oml.utils.remote_storage import download_folder_from_remote_storage
 
 
-def get_mock_images_dataset(
-    dataset_root: Union[str, Path] = MOCK_DATASET_PATH,
+def _get_mock_dataset(
+    dataset_local_folder: Union[str, Path],
+    dataset_remote_folder: Union[str, Path],
+    dataset_md5: str,
+    dataset_gdrive_url: str,
+    df_name: str,
     check_md5: bool = True,
-    df_name: str = "df.csv",
     global_paths: bool = False,
 ) -> Tuple[pd.DataFrame, pd.DataFrame]:
     """
-    Function to download mock dataset which is already prepared in the required format.
+    Downloads and prepares a mock dataset in the required format.
 
     Args:
-        dataset_root: Path to save the dataset
-        check_md5: Set ``True`` to check md5sum
-        df_name: Name of csv file for which output DataFrames will be returned
-        global_paths: Set ``True`` to cancat paths and ``dataset_root``
-
-    Returns: Dataframes for the training and validation stages
-
+        dataset_local_folder: The directory where the dataset will be saved.
+        dataset_remote_folder: The remote directory from which the dataset will be downloaded.
+        dataset_md5: The MD5 checksum used to validate the dataset.
+        dataset_gdrive_url: The Google Drive URL for the dataset download as a fallback option.
+        df_name: The name of the CSV file from which the output DataFrames will be generated.
+        check_md5: If True, validates the dataset using an MD5 checksum. Default is True.
+        global_paths: If True, concatenates the paths in the dataset with the dataset_local_folder. Default is False.
+
+    Returns:
+        A tuple containing two DataFrames:
+            - The first DataFrame is for the training stage.
+            - The second DataFrame is for the validation stage.
+
+    Raises:
+        Exception: If the downloaded dataset is invalid.
     """
-    dataset_root = Path(dataset_root)
+    dataset_local_folder = Path(dataset_local_folder)
+    dataset_remote_folder = Path(dataset_remote_folder)
+    dataset_md5 = dataset_md5 if check_md5 else None
 
-    if not check_exists_and_validate_md5(dataset_root, MOCK_DATASET_MD5 if check_md5 else None):
+    if not check_exists_and_validate_md5(dataset_local_folder, dataset_md5):
         try:
             print("Downloading from oml.daloroserver.com")
-            download_folder_from_remote_storage(remote_folder=MOCK_DATASET_PATH.name, local_folder=str(dataset_root))
-            assert check_exists_and_validate_md5(dataset_root, MOCK_DATASET_MD5 if check_md5 else None)
+            download_folder_from_remote_storage(
+                remote_folder=dataset_remote_folder.name, local_folder=str(dataset_local_folder)
```

Comment:
I think there is a confusing logic in `dataset_remote_folder` argument and the way we use it. Later on, in the domain-specific function we pass **local** path here (for example, `MOCK_DATASET_PATH`, which even starts from local cache folder), after which we only keep the last part of the path calling `.name()` and use it to navigate in server directories.

IMHO, the better option is not to transform `dataset_remove_folder` inside this function and have this logic outside like this:

images:
```
    return _get_mock_dataset(
        dataset_local_folder=dataset_root,
        dataset_remote_folder=MOCK_DATASET_PATH.name,  # <<<< HERE
        dataset_md5=MOCK_DATASET_MD5,
        dataset_gdrive_url=MOCK_DATASET_URL_GDRIVE,
        df_name=df_name,
        check_md5=check_md5,
        global_paths=global_paths,
    )
```
and audio:

```
return _get_mock_dataset(
        dataset_local_folder=dataset_root,
        dataset_remote_folder=MOCK_AUDIO_DATASET_PATH.name,  # <<<<<< HERE
        dataset_md5=MOCK_AUDIO_DATASET_MD5,
        dataset_gdrive_url=MOCK_AUDIO_DATASET_URL_GDRIVE,
        df_name=df_name,
        check_md5=check_md5,
        global_paths=global_paths,
    )
```

Reply:
fixed